StencilFrontend.md

```diff
@@ -0,0 +1,353 @@
+
+If you have not completed the [FoodTrackerBackend](https://github.com/IBM/FoodTrackerBackend) tutorial go to the [CompletedFoodTracker](https://github.com/IBM/FoodTrackerBackend/tree/CompletedFoodTracker) branch and follow the [README](https://github.com/IBM/FoodTrackerBackend/blob/CompletedFoodTracker/README.md) instructions.
+
+## Using A Stencil Template
+The FoodTracker application is taken from the Apple tutorial for building your first iOS application. In the [FoodTrackerBackend](https://github.com/IBM/FoodTrackerBackend) tutorial, we created a server and connected it to the iOS application. This means that when meals are added they are posted to the server, a user can then view these meals on [localhost:8080/meals](http://localhost:8080/meals). When you perform a HTTP `GET` request to the server, you are returned a JSON array which contains your stored meals. To present this data in a more appealing way we will use [Stencil](https://github.com/kylef/Stencil). Stencil is a template language for Swift which allows you to embed code into a HTML document to create a dynamic web page.
+
+### Create the Stencil File
+To use templates with a Kitura router, you must create a `.stencil` template file which describes how to embed code into the returned HTML. By default, the Kitura router gets the template files from the `./Views/` directory in the directory where Kitura runs.
+
+1. In the terminal, change into the FoodServer directory:
+```
+cd ~/FoodTrackerBackend/FoodServer
+```
+2. Create the Views folder and change into it:
+```
+mkdir Views
+cd Views
+```
+3. Create a `FoodTemplate.stencil` file.
+We will use terminal commands for file creation but you can use a text editor of your choice.
+```
+cat > FoodTemplate.stencil
+```
+**Note:** You can exit `cat` using `CTRL+D`
+
+4. Add the following Stencil template code to display the meals:
+
+```
+<html>
+There are {{ meals.count }} meals. <br />
+
+{% for meal in meals %}
+    - {{ meal.name }} with rating {{ meal.rating }}. <br />
+    <br />
+{% endfor %}
+</html>
+```
+5. Save the file and exit your text editor. For `cat` use the shortcut:  `CTRL+D`
+
+When the above code is rendered, it will display the total number of meals in your meal store, it will then loop through the meal store displaying each meal name and rating. For more details about Stencil templates see the [Stencil User Guide](http://stencil.fuller.li/en/latest/).
+
+
+### Add the Kitura-StencilTemplateEngine dependency
+[Kitura-TemplateEngine](https://github.com/IBM-Swift/Kitura-TemplateEngine) is a template engine abstraction layer which works alongside a specific template engine, such as [Kitura-StencilTemplateEngine](https://github.com/IBM-Swift/Kitura-StencilTemplateEngine), to allow a user to easily use a template engine in Swift. We need to add the `Kitura-StencilTemplateEngine` dependency to our `Package.swift` file.
```

Comment:
Do we need to talk about Kitura-TemplateEngine here as we're not directly using it?